src/libraries/ReceiverWeights.sol

```diff
@@ -79,65 +74,36 @@ library ReceiverWeightsImpl {
         returns (
             address receiver,
             address receiverHint,
-            uint32 weightReceiver,
-            uint32 weightProxy
+            uint32 weight
         )
     {
         receiver = (prevReceiver == ADDR_ROOT) ? self.data[ADDR_ROOT].next : prevReceiverHint;
         while (receiver != ADDR_ROOT) {
-            weightReceiver = self.data[receiver].weightReceiver;
-            weightProxy = self.data[receiver].weightProxy;
+            weight = self.data[receiver].weight;
             receiverHint = self.data[receiver].next;
-            if (weightReceiver != 0 || weightProxy != 0) break;
+            if (weight != 0) break;
             receiver = receiverHint;
         }
     }
 
-    /// @notice Checks if the list is fully zeroed and takes no storage space.
-    /// It means that either it was never used or that
-    /// it's been pruned after removal of all the elements.
-    /// @return True if the list is zeroed
-    function isZeroed(ReceiverWeights storage self) internal view returns (bool) {
-        return self.data[ADDR_ROOT].next == ADDR_ROOT;
-    }
-
     /// @notice Set weight for a specific receiver
     /// @param receiver The receiver to set weight
     /// @param weight The weight to set
     /// @return previousWeight The previously set weight, may be zero
-    function setReceiverWeight(
+    function setWeight(
         ReceiverWeights storage self,
         address receiver,
         uint32 weight
     ) internal returns (uint32 previousWeight) {
-        self.attachToList(receiver);
-        previousWeight = self.data[receiver].weightReceiver;
-        self.data[receiver].weightReceiver = weight;
-    }
-
-    /// @notice Set weight for a specific proxy
-    /// @param proxy The proxy to set weight
-    /// @param weight The weight to set
-    /// @return previousWeight The previously set weight, may be zero
-    function setProxyWeight(
-        ReceiverWeights storage self,
-        address proxy,
-        uint32 weight
-    ) internal returns (uint32 previousWeight) {
-        self.attachToList(proxy);
-        previousWeight = self.data[proxy].weightProxy;
-        self.data[proxy].weightProxy = weight;
-    }
-
-    /// @notice Ensures that the weight for a specific receiver is attached to the list
-    /// @param receiver The receiver whose weight should be attached
-    function attachToList(ReceiverWeights storage self, address receiver) internal {
         require(receiver != ADDR_ROOT, "Invalid receiver address");
+        // Ensure that the weight for a specific receiver is attached to the list
```

Comment:
Ah okay you removed the `attachToList` because it is only used once and we can save some gas.

Reply:
I doubt that it saves any gas, if the solidity optimizer isn't very different from regular compilers, it probably was inlining anyway :shrug: